bayboone/inference/module.py

```diff
@@ -0,0 +1,132 @@
+import numpy as np
+import pandas as pd
+import pymc3 as pm
+import matplotlib.pyplot as plt
+import arviz as az
+# Adapted from week 07 notebook
+
+def oscillation_model(num_neutrinos, num_nue):
+    '''
+    Creates a statistical model for predicting the oscillation parameters from microboone-like values
+    Inputs:
+    num_neutrinos: 
+        The number of muon neutrinos shot at the detector
+    num_nue:
+        The number of electron neutrinos detected
+    Returns:
+    osc_model: pymc3 model
+        the statistical model for our neutrino oscillations
+    '''
+    #Check that the data is reasonable
+    if (num_neutrinos < num_nue):
+        raise ValueError("Error: number of initial neutrinos cannot be less than number of nues observed")
+
+    osc_model = pm.Model()
+    with osc_model:
+    
+        # Priors for unknown model parameters
+        ss2t = pm.Uniform('sin^2_2theta', 0, 1)
+        dms = pm.Uniform('delta_m^2', 0, 1.0) #units of ev^2
+        # We don't know the exact energy or production point of each neutrino, so we draw from a truncated  gaussian (enforcing positive distance travelled and energy)
+        #L_over_E = pm.Normal('L_over_E', mu = 0.5, sigma = 0.1 ) #units of km/Gev
+        L = pm.TruncatedNormal('L', mu = 0.500, sigma = 0.05, lower = 0) #units of km
+        E = pm.TruncatedNormal('E', mu = 1.0, sigma = 0.05, lower = 0) #units of GeV
+
+        # Expected value from theory 
+        #P = pm.Deterministic('prediction', ss2t*(np.sin(dms*(1.27*L)/E))**2)
+        # Likelihood of observations
+        # measurements = pm.Binomial('nue_Flux', n=num_neutrinos, p=P, observed=num_nue)
+        
+        # In the large n limit, because the number of oscillations is low, we use a Poisson approximation
+        # Rate parameter calculated form theory
+        rate = pm.Deterministic('rate', num_neutrinos*ss2t*(np.sin(dms*(1.27*L)/E))**2)
+        
+        #Likelihood of observations
+        measurements = pm.Poisson('nue_flux', mu = rate, observed = num_nue)
+        
+    return osc_model
+
+def fit_model(data):
```

Comment:
Can we keep this consistent with the rest of the package and use N_numu, N_nue, rather that this data array?

Reply:
yup